### PR TITLE
chore: backport github action

### PR DESCRIPTION
**Pull request by SebastianElvis**
Part of https://github.com/babylonlabs-io/babylon/issues/110

This PR introduces the backport Github action. One can label a hotfix PR to `main` with `backport release/0.Y.x`, and this GH action will open a backport PR automatically.

### CODE REVIEW

File: .github/workflows/backport.yml
Comment (gitferry) [inline .github/workflows/backport.yml:5]:
```suggestion
#   - Then label it with `backport production`.
```

**Review by gitferry — APPROVED** [review]
Great!

**Comment by KonradStaniec** [comment]
oh this loos great! 

Does the backport happens through pr which must be approved or it is happens by commit cherry pick and merge to backported branch ?

**Review by RafilxTenfen — APPROVED** [review]
:rocket:

**Review by Lazar955 — APPROVED** [review]

**Comment by SebastianElvis** [comment]
> oh this loos great!
> 
> Does the backport happens through pr which must be approved or it is happens by commit cherry pick and merge to backported branch ?

As per https://github.com/babylonlabs-io/babylon/pull/149/files#diff-1b2ad8b0da1ad4fac7eb4ef8cdeed82b48d00a80eb531a6f31db11f73182f491R24-R34 I think it happens as soon as a PR with backport label is merged. Also we require approval for merging PRs.

**Pull request merged**
